httpcache.go

```diff
@@ -51,7 +55,48 @@ func CachedResponse(c Cache, req *http.Request) (resp *http.Response, err error)
 	}
 
 	b := bytes.NewBuffer(cachedVal)
-	return http.ReadResponse(bufio.NewReader(b), req)
+	returnResponse, err := http.ReadResponse(bufio.NewReader(b), req)
+	if err != nil {
+		return nil, fmt.Errorf("error loading response from cache: %s\n", err.Error())
+	}
+
+	rangeRaw := req.Header.Get("range")
+	if rangeRaw != "" {
+		tmp := strings.Split(rangeRaw, rangeTypeSeparator)
+		// standard format is bytes=START-END
+		rangetype, rangevalue := tmp[0], tmp[1]
+		if rangetype != "bytes" {
+			fmt.Printf("range type %s not supported\n", rangetype)
+			return returnResponse, nil
+		}
+		// we need to read all the body now, close it, and replace it with another reader
+		// as there is currently no way of "resetting" a Body
+		body, err := ioutil.ReadAll(returnResponse.Body)
+		if err != nil {
+			fmt.Printf("error reading cached response body: %s\n", err.Error())
+			return returnResponse, nil
+		}
+		returnResponse.Body.Close()
+		var rangedRequestStart, rangedRequestEnd int64
```

Comment:
the first one is spot on, sorry i missed that
the idea was to have range before (from the header) and ranged after i split/validated it, but i notice that it would still not be enough, will rework